source/extensions/filters/udp/dns_filter/dns_parser.cc

```diff
@@ -193,8 +193,11 @@ bool DnsMessageParser::parseDnsObject(DnsQueryContextPtr& context,
     }
   } while (!done);
 
-  if (!context->header_.flags.qr && context->header_.answers) {
-    ENVOY_LOG(debug, "Answer records present in query");
+  // Only QR == 0 and questions without any answer RRs are expected
+  if (!(context->header_.flags.qr == 0 && context->header_.answers == 0 &&
+        context->header_.authority_rrs == 0 && context->header_.additional_rrs == 0)) {
+    ENVOY_LOG(debug, "One or more of Answer, Authority, and Additional RRs present in the query",
+              "Inverse query is not supported");
```

Comment:
Do you eventually want to add a stat here? Might be hard to debug this in practice without it. Fine to do as a follow up. If so maybe add a TODO.

Also, can you add a release note for this? It's not clear to me what was the behavior before and what the behavior is now.

Reply:
Will print answers's count with `static_cast` as well. for stat, do you mean the above or statistic?

Previously, the parser will return *true* when client sends a malformed message with question count == 1 and answer count == 1, which is not expected. The new behavior will only accept query with Question count == 1, and all other answer RRs count ==0. Otherwise, a format error code will be sent back.

Reply:
I mean that I think top use this in production this filter needs actual metrics that can be monitored, but I suppose that can be added later.

Reply:
Thank you for clarifying. We can add the stat/metric later. This kind of message is not expected before as well for production use.